packages/client/src/monaco-languages.ts

```diff
@@ -190,7 +240,17 @@ export class MonacoLanguages implements Languages {
     protected createCodeActionProvider(provider: CodeActionProvider): monaco.languages.CodeActionProvider {
         return {
             provideCodeActions: async (model, range, context, token) => {
-                const params = this.m2p.asCodeActionParams(model, range, context);
+                const allDiagnostics: Diagnostic[] = [];
+                for (const diagnostic of this.extHostDiagnostics.getDiagnostics(model.uri.toString())) {
+                    if (range.intersectRanges(this.p2m.asRange(diagnostic.range))) {
+                        const newLen = allDiagnostics.push(diagnostic);
+                        if (newLen > 1000) {
```

Comment:
Do mind putting this number (1000) into a static constant?

Reply:
of course: 164f4206cced7af444f49d274248f6b656ceb5ba

I'm not sure about the constant name, it comes from the [vscode code itself](https://github.com/microsoft/vscode/blob/e9393b096d90a5ac49b1528d1ec60483fc6b993e/src/vs/workbench/api/common/extHostLanguageFeatures.ts#L377)

That's why I didn't put it in a constant in the first place

Reply:
What about mentioning this in a comment (including the link)

Reply:
sure: 00cfc7f1c38a4738df1bcb6a441923f8b2014bea